### ISSUE TITLE
Incorrect error parsing for markdownlint-cli checker (markdownlint-cli 0.24.0)

### ISSUE THREAD

**Issue by czikus** [flycheck/flycheck#1837]
# Bug description

This is a continuation of #1647, which resurfaced with markdown-cli 0.24.0. It seems that now, some errors report the column, and some don't, e.g. :

```
test.md:22:2 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
test.md:27:4 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
test.md:28:15 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
test.md:33 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 2; Actual: 3]
test.md:74 MD003/heading-style/header-style Heading style [Expected: setext; Actual: atx_closed]
```

## Steps to reproduce
1. `npm install -g markdownlint-cli`
1. Open a markdown file with different types of errors (see above for examples)
1. Open the flycheck error list and see that some errors are assigned an incorrect location

## System configuration
Emacs configuration:
- [ ] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [x] Doom Emacs
- [ ] Other shared configuration

**Comment by cpitclaudel** [comment]
Thanks a lot for the report.  Do you feel comfortable updating the error pattern?  Otherwise I can help.

**Comment by czikus** [comment]
This seems to work just fine:

```
((error line-start
          (file-name) ":" line ":" column " " (id (one-or-more (not (any space))))
          " " (message) line-end)
   (error line-start
          (file-name) ":" line " " (id (one-or-more (not (any space))))
          " " (message) line-end))
```

**Comment by cpitclaudel** [comment]
Great! Probably even better to have a single pattern with `(? ":" column)`; can you make a PR?

**Comment by czikus** [comment]
Could you please make a patch? Cannot do the PR right now.

**Comment by czikus** [comment]
Here's the PR. Please review.
